bpf/process/bpf_process_event.h

```diff
@@ -288,4 +288,31 @@ set_in_init_tree(struct execve_map_value *curr, struct execve_map_value *parent)
 		DEBUG("%s: nspid=1", __func__);
 	}
 }
+
+#ifdef __LARGE_BPF_PROG
+FUNC_INLINE struct execve_map_value *
+event_find_curr_probe(struct msg_generic_kprobe *msg)
+{
+	struct task_struct *task = (struct task_struct *)get_current_task();
+	struct execve_map_value *curr;
+
+	curr = &msg->curr;
+	curr->key.pid = BPF_CORE_READ(task, tgid);
+	curr->key.ktime = ktime_get_ns();
+	curr->nspid = get_task_pid_vnr_by_task(task);
+
+	get_current_subj_caps(&curr->caps, task);
+	get_namespaces(&curr->ns, task);
+	set_in_init_tree(curr, NULL);
+
+	read_exe((struct task_struct *)get_current_task(), &msg->exe);
+	return curr;
+}
+#else
+FUNC_INLINE struct execve_map_value *
+event_find_curr_probe(struct msg_generic_kprobe *msg)
+{
+	return NULL;
+}
+#endif
```

Comment:
what's the benefit of doing this instead of writing the function unconditionally and making the calling conditional to LARGE PROGS? Wouldn't that be more explicit that non large prog do not support that?

Reply:
well I think it's logically isolated piece of code which should be in function and is different for `__LARGE_BPF_PROG` define, so it makes sense to me to do the above

also I think it's better/readable/more manageble than having ifdefs inside the code